qa/wildfly/build.gradle

```diff
@@ -0,0 +1,190 @@
+    doFirst {
+        final File script = new File(project.buildDir, "wildfly/wildfly.killer.sh")
+        script.setText(
+                ["function shutdown {",
+                 "  ${stopWildflyCommand}",
+                 "}",
+                 "trap shutdown EXIT",
+                 // will wait indefinitely for input, but we never pass input, and the pipe is only closed when the build dies
+                 "read line"].join('\n'), 'UTF-8')
+        final ProcessBuilder pb = new ProcessBuilder("bash", script.absolutePath)
+        pb.start()
+    }
+    doLast {
+        final ProcessBuilder pb =
+                new ProcessBuilder("${wildflyInstall}/bin/standalone.sh", "-Djboss.socket.binding.port-offset=${portOffset}")
+        final Process process = pb.start()
+        new BufferedReader(new InputStreamReader(process.getInputStream())).withReader { br ->
+            String line
+            while ((line = br.readLine()) != null) {
+                if (line.matches(".*WildFly Full \\d+\\.\\d+\\.\\d+\\.Final \\(WildFly Core \\d+\\.\\d+\\.\\d+\\.Final\\) started.*")) {
+                    break
+                }
+            }
+        }
+    }
+}
+
+task configureTransportClient(type: LoggedExec) {
+    dependsOn startWildfly
+    commandLine "${wildflyInstall}/bin/jboss-cli.sh",
+            "--controller=localhost:${managementPort}",
+            "--connect",
+            "--command=/system-property=elasticsearch.properties:add(value=\${jboss.server.config.dir}/elasticsearch.properties)"
+}
+
+task stopWildfly(type: LoggedExec) {
+    commandLine stopWildflyCommand.split(' ')
+}
+
+if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+    integTestRunner.dependsOn(configureTransportClient)
+    final TaskExecutionAdapter logDumpListener = new TaskExecutionAdapter() {
+        @Override
+        void afterExecute(final Task task, final TaskState state) {
+            if (state.failure != null) {
+                final File logFile = new File(wildflyInstall, "standalone/log/server.log")
+                println("\nWildfly server log (from ${logFile}):")
+                println('-----------------------------------------')
+                final Stream<String> stream = Files.lines(logFile.toPath(), StandardCharsets.UTF_8)
+                try {
+                    for (String line : stream) {
+                        println(line)
+                    }
+                } finally {
+                    stream.close()
+                }
+                println('=========================================')
+            }
+        }
+    }
+    integTestRunner.doFirst {
+        project.gradle.addListener(logDumpListener)
+    }
+    integTestRunner.doLast {
+        project.gradle.removeListener(logDumpListener)
+    }
+    integTestRunner.finalizedBy(stopWildfly)
+} else {
+    integTest.enabled = false
+}
+
+check.dependsOn(integTest)
+
+test.enabled = false
+
+dependencyLicenses.enabled = false
```

Comment:
I discussed this with @rjernst. We are going to create an examples directory and we can move the generation of the war to a subproject there. This test subproject can depend on that artifact. That will enable this subproject to depend apply standalone-rest-test. This will be done a follow-up.